src/coreclr/jit/emitxarch.cpp

```diff
@@ -677,6 +677,116 @@ bool emitter::AreUpper32BitsSignExtended(regNumber reg)
 }
 #endif // TARGET_64BIT
 
+bool emitter::IsRedundantCmp(emitAttr size, regNumber reg1, regNumber reg2)
+{
+    // Only allow GPRs.
+    // If not a valid register, then return false.
+    if (!genIsValidIntReg(reg1))
+        return false;
+
+    if (!genIsValidIntReg(reg2))
+        return false;
+
+    // Only consider if safe
+    //
+    if (!emitCanPeepholeLastIns())
+    {
+        return false;
+    }
+
+    bool result = false;
+
+    //instInfo
+    emitPeepholeIterateLastInstrs([&](instrDesc* id) {
+        switch (id->idIns())
+        {
+            case INS_seta:
```

Comment:
I don't really want to specify specific instructions - I'm just doing this for now.

Reply:
Where did you get this list from?

Reply:
I'm just aware of the set and jmp instructions and we have the list defined, so I just copied it from there.